frontend/src/lib/components/buttonAlt/Button.svelte

```diff
@@ -0,0 +1,127 @@
+  // This is called variant instead of type because that clashes with an attribute of <a>
+  export let variant: $$Props['variant'] = 'normal';
+  export let icon: $$Props['icon'] = variant === 'main' ? 'next' : null;
+  export let iconPos: $$Props['iconPos'] = variant === 'main' ? 'right' : 'left';
+  export let color: $$Props['color'] = 'primary';
+
+  // Check iconPos
+  if (variant === 'main' && (iconPos === 'top' || iconPos === 'bottom')) {
+    iconPos = 'right';
+  }
+
+  // Element type
+  const tagName = $$restProps['href'] ? 'a' : 'button';
+
+  // Build classes
+  // 1. Base classes
+  let classes =
+    'btn min-h-touch min-w-touch h-auto flex items-center justify-center gap-y-6 gap-x-4';
+
+  // 2. Variant-defined classes
+  classes += variant === 'main' ? ' w-full max-w-md' : ' btn-ghost';
+
+  // 3. Icon position
+  switch (iconPos) {
+    case 'top':
+      classes += ' flex-col';
+      break;
+    case 'bottom':
+      classes += ' flex-col-reverse';
+      break;
+    case 'left':
+      classes += ' flex-row';
+      break;
+    case 'right':
+      classes += ' flex-row-reverse';
+      break;
+  }
+
+  // 4. Set color
+  if (color) {
+    // For the main button type we can use the btn-primary etc. classes,
+    // for the others, we just set the text color
+    classes += ` ${variant === 'main' ? 'btn' : 'text'}-${color}`;
+  }
+
+  // 5. Merge classes into $$restProps
+  $$restProps.class = `${classes} ${$$restProps.class ?? ''}`;
+
+  // 6. Finally, define the class for the text label
+  let labelClass: string;
+  if (variant === 'main') {
+    labelClass = 'flex-grow text-center';
+    if (icon) {
+      // If an icon is used, add left or right margin so that the text is
+      // nicely centered: ml/r is calculated so that it is the sum of the
+      // gap (4) and icon widths (24) = 28/16 rem
+      labelClass += iconPos === 'right' ? ' ml-[1.75rem]' : ' mr-[1.75rem]';
+    }
+  } else if (icon && (iconPos === 'top' || iconPos === 'bottom')) {
+    // We use the small-label class only in vertical buttons with an icon.
+    // The color needs to be separately applied here, bc small-label sets
+    // the text color to secondary
+    labelClass = `small-label text-${color}`;
+  }
+</script>
+
+<!--
+@component
+A component for buttons that mostly contain text and an icon. Use the
+`variant` prop to specify the button type. When using an `icon`, use
+`iconPos` to set the position of the icon relative to the text.
+
+- `main`: A large, prominent button that is used for the main action of
+  the page. In general, there should only be one of these on a page.
+- `icon`: A button containing only an icon. Note that you still need to
+  provide the `text` property, which will be used as the Aria label for
+  the button.
+- `normal`: The default button type, which usually consists of an icon
+  and text. The styling for these uses DaisyUI's `btn-ghost` class, i.e.
+  the button does not have a background color.
+
```

Comment:
can we mention that people can change the color of the text using the  `text-{COLOR}` utility classes?